text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,678 @@
+  maintainers.  (It may still be a niche target, but the host tools must not be
+  exclusively useful for an inherently closed group.) This requirement will be
+  evaluated independently from the corresponding tier 2 requirement.
+  - The requirement to provide "direct value" means that it does not suffice to
+    argue that having host tools will help the target's maintainers more easily
+    provide the target to others. The tools themselves must provide value to
+    others.
+- There must be a reasonable expectation that the host tools will be used, for
+  purposes other than to prove that they can be used.
+- The host tools must build and run reliably in CI, though they may or may not
+  pass tests.
+- Building host tools for the target must not take substantially longer than
+  building host tools for other targets.
+- The host tools must provide a substantively similar experience as on other
+  targets, subject to reasonable target limitations.
+  - Adding a substantively different interface to an existing tool, or a
+    target-specific interface to the functionality of an existing tool,
+    requires design and implementation approval (e.g. RFC/MCP) from the
+    appropriate approving teams for that tool.
+    - Such an interface should have a design that could potentially work for
+      other targets with similar properties.
+    - This should happen separately from the review and approval of the target,
+      to simplify the target review and approval processes, and to simplify the
+      review and approval processes for the proposed new interface.
+  - By way of example, a target that runs within a sandbox may need to modify
+    the handling of files, tool invocation, and similar to meet the
+    expectations and conventions of the sandbox, but must not introduce a
+    separate "sandboxed compilation" interface separate from the CLI interface
+    without going through the normal approval process for such an interface.
+    Such an interface should take into account potential other targets with
+    similar sandboxes.
+- If the host tools for the platform would normally be expected to be signed or
+  equivalent (e.g. if running unsigned binaries or similar involves a
+  "developer mode" or an additional prompt), it must be possible for the Rust
+  project's automated builds to apply the appropriate signature process,
+  without any manual intervention by either Rust developers, target
+  maintainers, or a third party.  This process must meet the approval of the
+  infrastructure team.
+  - This process may require one-time or semi-regular manual steps by the
+    infrastructure team, such as registration or renewal of a signing key. Any
+    such manual process must meet the approval of the infrastructure team.
+  - This requirement exists to ensure that Rust builds, including nightly
+    builds, can meet the necessary requirements to allow users to smoothly run
+    the host tools.
```

Comment:
I've updated this specific requirement to note that signing processes and similar may require a legal agreement, and that such an agreement may be revocable or involve a nominal fee, but must not be otherwise onerous. (It's not reasonable, for instance, for a signing-related agreement to require that members of the infrastructure team not work on competing systems or for competing companies.)

Tier 2 intentionally doesn't *require* that the target allow unsigned binaries; that's a tier 1 requirement. The intention is that a target with a hard requirement for signed binaries may be allowed at tier 2, but is not allowed at tier 1.